schema/testdata.go

```diff
@@ -231,7 +233,10 @@ func (tg *TestDataGenerator) Generate(table *Table, opts GenTestDataOptions) arr
 
 	var records []arrow.Record
 	for j := 0; j < opts.MaxRows; j++ {
-		tg.counter++
+		if !opts.DuplicateData {
+			tg.counter++
```

Comment:
What kind of dup data is generated when`tg.counter` isn't incremented?

Reply:
All `id` columns are the same:
```
	// special case for auto-incrementing id column, used for to determine ordering in tests
	if arrow.IsInteger(dataType.ID()) && colName == "id" {
		return `` + strconv.Itoa(tg.counter) + ``
	}
```